### PR TITLE
Activity Panel: Inbox: When the user updates WooCommerce, add a note alerting that the store notice setting has moved

### PR DESCRIPTION
**Pull request by allendav**
See #176 

Hooks into WooCommerce update. On updating WooCommerce, adds a "store notice setting has moved" note to the merchant's inbox if one isn't already in their inbox.

### Screenshots

screen shot 2018-10-23 at 10 31 28 am

### Detailed test instructions:

- To avoid having to update WooCommerce (which will be a pain), instead, modify class-wc-install.php in woocommerce as follows. (Note: check_version is called on every WooCommerce request.)
- Then, visit `/wp-admin/admin.php?page=wc-admin#/` and look in your inbox.
- Note: All actions will be enabled in a subsequent PR.

```
	/**
	 * Check WooCommerce version and run the updater is required.
	 *
	 * This check is done on all requests and runs if the versions do not match.
	 */
	public static function check_version() {
		//if ( ! defined( 'IFRAME_REQUEST' ) && version_compare( get_option( 'woocommerce_version' ), WC()->version, '<' ) ) {
		//	self::install();
			do_action( 'woocommerce_updated' );
		//}
	}
```

Note: I also DRYd includes/class-wc-admin-notes-new-sales-record.php a little in this PR.

### CODE REVIEW

**Review by timmyc — APPROVED** [review]
One minor comment on link target, but testing great for me! Pre-approving.

File: includes/class-wc-admin-notes-settings-notes.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		$note->set_content( __( 'It can now be found in the Customizer.', 'wc-admin' ) );
+		$note->set_content_data( (object) array() );
+		$note->set_type( WC_Admin_Note::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_icon( 'info' );
+		$note->set_name( $name );
+		$note->set_source( 'wc-admin' );
+		$note->add_action(
+			'open-customizer',
+			__( 'Open Customizer', 'wc-admin' ),
+			'?page=wc-admin'
```
[the remark this answers is not in the document]
Comment (allendav) [reply includes/class-wc-admin-notes-settings-notes.php]: LOL yep

[the remark this answers is not in the document]
Comment (timmyc) [reply includes/class-wc-admin-notes-settings-notes.php]: :trollface:

**Comment by allendav** [comment]
@timmyc - thanks for catching that the action URL was incorrect - I have fixed that (I completely forgot to circle back to that) - in the process I went ahead and enabled the buttons to make testing easier - they just aren't tracked yet

Please give this another spin (same testing instructions as above - but first use the Test Bench Tool to delete all old notices since they have the old "URL" and will not be regenerated otherwise - https://github.com/automattic/wc-admin-notes-test-bench )

@LevinMedia - you'll see a CSS change in here - without it the button label is white and pretty unreadable. Please let me know if I you would rather a different value for the label color.

**Comment by LevinMedia** [comment]
Heya @allendav thanks for the ping on this! I cross referenced the color with `.wp-core-ui .button, .wp-core-ui .button-secondary` which appears elsewhere in wp-admin, and that's using `#555` for it's text. 

screen shot 2018-10-25 at 8 43 56 am

Would it be possible to use the button directly from core in this component to keep things consistent?

**Comment by allendav** [comment]
@LevinMedia - changing the component classes to match core's would probably make this a little brittle - if core changes styles it could have unexpected visual impact - so... I'll copy the #555555 to our style sheets instead

**Pull request merged**